client/components/update-post-status/index.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import UpdateTemplate from './update-template';
+import PostActions from 'lib/posts/actions';
+import analytics from 'lib/analytics';
+
+// TODO(mcsf): use i18n#localize?
+const strings = {
+	page: {
+		deleteWarning: translate( 'Delete this page permanently?' ),
+		deleted: translate( 'Page Deleted' ),
+		deleting: translate( 'Deleting Page' ),
+		error: translate( 'Error' ),
+		restored: translate( 'Page Restored' ),
+		restoring: translate( 'Restoring Page' ),
+		trashed: translate( 'Moved to Trash' ),
+		trashing: translate( 'Trashing Page' ),
+		undo: translate( 'undo?' ),
+		updated: translate( 'Updated' ),
+		updating: translate( 'Updating Page' ),
+	},
+	post: {
+		deleteWarning: translate( 'Delete this post permanently?' ),
+		deleted: translate( 'Post Deleted' ),
+		deleting: translate( 'Deleting Post' ),
+		error: translate( 'Error' ),
+		restored: translate( 'Restored' ),
+		restoring: translate( 'Restoring' ),
+		trashed: translate( 'Moved to Trash' ),
+		trashing: translate( 'Trashing Post' ),
+		undo: translate( 'undo?' ),
+		updated: translate( 'Updated' ),
+		updating: translate( 'Updating Post' ),
+	},
+};
+
+const updatePostStatus = ( WrappedComponent ) =>
+	class UpdatePostStatus extends Component {
+		static displayName = `UpdatePostStatus(${
+			WrappedComponent.displayName || WrappedComponent.name || ''
+		})`;
+
+		state = {
+			updated: false,
+			updatedStatus: null,
+			previousStatus: null,
+			showMoreOptions: false,
+			showPageActions: false,
+		};
+
+		getType() {
+			return this.props.page ? 'page' : 'post';
+		}
+
+		buildUpdateTemplate = () => {
+			if ( ! this.state.updated ) {
+				return;
+			}
+
+			return <UpdateTemplate
+				post={ this.props.post || this.props.page }
+				previousStatus={ this.state.previousStatus }
+				resetToPreviousState={ this.resetToPreviousState }
+				status={ this.state.updatedStatus }
+				strings={ strings[ this.getType() ] }
+			/>;
+		}
```

Comment:
Either way works for me tbh. The babel transform example does it that way, since the semis are added afterwards. I'd just like us to be consistent here.